### ISSUE TITLE
Hostname setting is a bit confusing

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-management#30]
The `hostname` setting is used to configure the interface the server binds to. I.e. in many cases you might want to set it to `0.0.0.0` to bind to all interfaces.

Maybe the setting could be renamed to `interface` to clarify the intention.

(One example of the confusion on the ML: https://groups.google.com/d/topic/akka-user/jbBor1SuXhs/discussion.)

**Comment by speedcom** [comment]
+1 to this. I've set the hostname to `0.0.0.0` as a default one.

**Comment by patriknw** [comment]
Couldn’t it be a security consern to use 0.0.0.0 by default?

**Comment by ktoso** [comment]
Perhaps would be best to be explicit about it and not have a default there, if no one set it, we fail with an exception explaining how to set it. And also add a param to start with it perhaps in case one wants to start it programatically?

**Comment by patriknw** [comment]
Perhaps bind it to the same hostname as akka-remote by default?

**Comment by RomanIakovlev** [comment]
I think current default (localhost) is the best choice. Or like @ktoso suggests, not have the default at all.

**Comment by dhpiggott** [comment]
I think the current documentation is slightly misleading. From https://developer.lightbend.com/docs/akka-management/current/akka-management.html#basic-configuration:
> You can configure hostname and port to use for the HTTP Cluster management by overriding the following:
 ```
akka.management.http.hostname = "127.0.0.1"
akka.management.http.port = 19999
 ```
> However, those are the values by default.

I've just spent an embarrassingly long amount of time trying to figure out why I was getting connection refused errors when I was calling the daemon from within the container it was running in.

It turned out to be because the the daemon was _not_ binding to 127.0.0.1 - instead binding to 172.18.0.2:

```
analytics_1     | 2018-01-31 22:15:58,820 INFO  akka.management.AkkaManagement AkkaManagement(akka://liquidity) - Including HTTP management routes for ClusterHttpManagement
analytics_1     | 2018-01-31 22:15:58,880 INFO  akka.management.AkkaManagement AkkaManagement(akka://liquidity) - Binding Akka Management (HTTP) endpoint to: 172.18.0.2:19999
analytics_1     | 2018-01-31 22:15:59,824 INFO  akka.management.AkkaManagement AkkaManagement(akka://liquidity) - Bound Akka Management (HTTP) endpoint to: 172.18.0.2:19999
```

Because the main docs had said the default is 127.0.0.1, I had not added an override.

I then checked https://github.com/akka/akka-management/blob/master/management/src/main/resources/reference.conf#L13 and saw this, which does explain the behavior I was seeing:
> InetAddress.getLocalHost.getHostAddress is used not overriden or empty

I propose simply updating the main docs to simply state that the default is "InetAddress.getLocalHost.getHostAddress, which may or may not be 127.0.0.1".

I should be able to prepare a PR if that would be agreeable.

**Comment by ktoso** [comment]
You're right, please send in a PR adjusting the docs.
Sorry that they're not to our usual standard yet, it's early days for this project still.

Thanks for offering to help!

**Comment by dhpiggott** [comment]
Thanks for the quick confirmation - if it saves one person from the time-sink I ended up in, it'll have been worth it! I've raised a PR. Having done that, I then had a thought about this issue itself.

> Couldn’t it be a security consern to use 0.0.0.0 by default?

My thinking is the same - better to have to opt-in to that.

Alternative idea, the effect of which would be to basically preserve the current behavior (in most environments), but which would allow #127 to be closed/reverted, and would make things Just Work for the environment I had the above problem in:

My understanding is that akka-remote specifically _does_ need a host_name_, because it compares the hostname in the actor ref of incoming messages against the configured hostname in order to be able to drop mis-delivered messages.

I'm wondering if that's why the current value is `InetAddress.getLocalHost.getHostAddress` - i.e. was the config here modelled on akka-remote?

The difference I see is that here there is no need for a hostname - an interface is all that's required. With that in mind, perhaps `InetAddress.getLoopbackAddress` would have been (and would now be) more appropriate? From https://docs.oracle.com/javase/9/docs/api/java/net/InetAddress.html#getLoopbackAddress():

> The InetAddress returned will represent the IPv4 loopback address, 127.0.0.1, or the IPv6 loopback address, ::1. The IPv4 loopback address returned is only one of many in the form 127.*.*.*

Added benefit: no DNS resolution involved (unlike `InetAddress.getLocalHost.getHostAddress`).
